server/src/main/scala/io/delta/standalone/internal/DeltaSharingCDCReader.scala

```diff
@@ -139,13 +140,28 @@ class DeltaSharingCDCReader(val deltaLog: DeltaLogImpl, val conf: Configuration)
       } else {
         require(timestampKey == DeltaDataSource.CDF_END_TIMESTAMP_KEY)
         // For ending timestamp the version should be before the provided timestamp.
-        Some(history.getActiveCommitAtTime(ts).version)
+        Some(getActiveCommitAtTime(ts).version)
       }
     } else {
       None
     }
   }
 
+  /**
+   * A wrapper around DeltaHistory manager method.
+   * The method implicitly validates the timestamp arg, so we remap its exception to CDF
+   * illegal arg exception.
+   */
+   private def getActiveCommitAtTime(ts: Timestamp): DeltaSharingHistoryManager.Commit = {
+     val commit = try {
```

Comment:
If you sync with the head, you don't need this change? The main concern is that history.getActiveCommitAtTime(ts) may throw other exceptions which may not be `DeltaCDFIllegalArgumentException`.

Or do you still prefer this change over what I did here? https://github.com/delta-io/delta-sharing/blob/main/server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala#L98

Neither is obviously better since both are not precisely catching and throwing the error.

Reply:
The scope of getActiveCommitAt is a lot smaller, and looking at the code it will most commonly only throw timestamp related exception. So it is safer imho.

The other one seems too broad, so we should perhaps remove that one until we find a better solution.